### ISSUE TITLE
Automatic text block creation on Return 

### ISSUE THREAD

**Issue by aggied** [kmturley/angular-sir-trevor#2]
Hey @kmturley saw you starred my old repo https://github.com/aggied/angular-trevor and I stumbled my way here. Great work! Very glad you're taking it down the ionic route as well. I'm actually using  https://github.com/yabwe/medium-editor for now, but would like to get back to a block editor. Perhaps we can get some adoption and get some help on the angular block editor idea...

What are you thoughts on automatic creation of a text block when the user presses Return? My particular use case requires me to store metadata about each paragraph, image, video -- hence the block style. It allows rich text but not a full WYSIWYG. Allowing `<br/>` or multiple `<p>` seems a little short of the original goal (at least for me anyway).  I had in my repo a couple different text block types: paragraph and subheading. There wasn't any need for any html inside the paragraph short of span-level.

**Comment by kmturley** [comment]
Hey, I think the block editor allows content types to be kept separate and the database clean of html. Example of types of blocks would be:
- text
- image
- video
- iframes (gdocs, slideshare etc)

Each block could then have further options which are available like:
- text (formatting: bold, underline, italic, paragraphs etc)
- image (caption, alt text, low/high res?)
- video (autoplay, controls etc)
- iframe (scrolling)

So my opinion would be that pressing return inside a text block would add new paragraphs, just like a regular textarea. I don't see a need to add a new text block after a text block?

If you want to collaborate on this, more than happy too!

**Comment by aggied** [comment]
Right! Think we're on the same page here. I've seen quite a few extension block types for [Sir Trevor](http://madebymany.github.io/sir-trevor-js/) -- it's a really slick way to create content and works well on mobile. 

The text blocks are the only challenging ones-- at least if there is any desire to have some features such as bold, underline, etc. inside a paragraph of otherwise normal text. I agree we don't want HTML in the content, but I think there should be an exception for inline tags -- this allows bold, underline, and italicized spans as well as hyperlinks. 

The value in adding a text block after a text block is keeping the paragraphs truly separated. I have a use case that requires me to store/display metrics about each paragraph. There are dozens of needs like this, such as QZ.com's ability to annotate paragraphs. 

> ...pressing return inside a text block would add new paragraphs, just like a regular textarea. 

I'm wondering if we're talking past each other on this. How are those new paragraphs handled? Are you thinking of relying on the textarea to place `<br/>` tags (HTML...)? And if not how without having HTML markup?

**Comment by kmturley** [comment]
Yeah, look at the demo:
http://kmturley.github.io/angular-sir-trevor/src/

You can see that when using a content editable div, hitting Return adds new paragraph < p > tags automatically! I decided to use this funcationality because it was clean and easy. But I didn't have the time to allow the user to highlight a portion of text and change the text element type (p, h1, h2, etc).

I think your idea of separating text elements out would work (less html in the database), but we would need to have a text block type (p, h1, h2, etc). I'm open to investigating this idea!

We could change the block types to be real dom elements e.g.
- h1, h2 etc
- p
- img
- video
- iframe

Then the text templates could somehow map the dom element dynamically:

```
<{{ type }} contenteditable="true">{{ block.content }}</{{ type }}>
```

And the directive could just use the dom element:

```
.directive('h1', [function () {
    'use strict';
    return {
        restrict: 'E',
        link: function (scope, element, attr, ctrl) {
            // do something
        }
    };
}])
```

Here is a quick example of how the basis of this could work:
http://jsfiddle.net/kmturley/enq9fpn4/
